File: leetcode-cn/1184.0_Distance_Between_Bus_Stops.py
'''
执行用时：36 ms, 在所有 Python3 提交中击败了68.51% 的用户
内存消耗：15.5 MB, 在所有 Python3 提交中击败了65.26% 的用户
通过测试用例：37 / 37
'''
class Solution:
    def distanceBetweenBusStops(self, distance: List[int], start: int, destination: int) -> int:
        s = sum(distance)
        if start > destination:
            start, destination = destination, start
        one = sum(distance[start: destination])
        return min(one, s - one)

'''
执行用时：40 ms, 在所有 Python3 提交中击败了40.33% 的用户
内存消耗：15.6 MB, 在所有 Python3 提交中击败了41.67% 的用户
通过测试用例：37 / 37
'''
class Solution:
    def distanceBetweenBusStops(self, distance: List[int], start: int, destination: int) -> int:
        s = sum(distance)
        if start > destination:
            start, destination = destination, start
        s1 = sum(distance[start: destination])
        return min(s1, s - s1)

File: java/1184.0_Distance_Between_Bus_Stops.java
/*
执行用时：0 ms, 在所有 Java 提交中击败了100.00% 的用户
内存消耗：40.8 MB, 在所有 Java 提交中击败了49.24% 的用户
通过测试用例：37 / 37
*/
class Solution {
    public int distanceBetweenBusStops(int[] distance, int start, int destination) {
        int d1 = 0, d2 = 0;
        int l = Math.min(start, destination);
        int r = Math.max(start, destination);
        for (int i = 0; i < distance.length; i++) {
            if (i >= l && i < r) {
                d1 += distance[i];
            } else {
                d2 += distance[i];
            }
        }
        return Math.min(d1, d2);
    }
}
